apps/asap-cli/src/scripts/algolia/performance/engagement.ts

```diff
@@ -0,0 +1,55 @@
+import { SearchIndex } from 'algoliasearch';
+
+import {
+  deletePreviousObjects,
+  getAllHits,
+  getBellCurveMetrics,
+  Hit,
+} from '../process-performance';
+
+type EngagementHit = Hit & {
+  eventCount: number;
+  totalSpeakerCount: number;
+  uniqueAllRolesCount: number;
+  uniqueKeyPersonnelCount: number;
+};
+
+export const processEngagementPerformance = async (index: SearchIndex) => {
+  const type = 'engagement' as const;
+
+  await deletePreviousObjects(index, type);
+
+  const getPaginatedHits = (page: number) =>
+    index.search<EngagementHit>('', {
+      filters: `__meta.type:"${type}"`,
+      attributesToRetrieve: [
+        'eventCount',
+        'totalSpeakerCount',
+        'uniqueAllRolesCount',
+        'uniqueKeyPersonnelCount',
+      ],
+      page,
+      hitsPerPage: 50,
+    });
+
+  const hits = await getAllHits<EngagementHit>(getPaginatedHits);
+
+  await index.saveObject(
+    {
+      events: getBellCurveMetrics(hits.map((hit) => hit.eventCount)),
```

Comment:
This has been merged https://github.com/yldio/asap-hub/pull/4356 so now it's possible to use `getPerformanceMetrics` instead :)
```suggestion
      events: getPerformanceMetrics(hits.map((hit) => hit.eventCount)),
```